cql/src/main/java/org/apache/cassandra/stargate/transport/internal/ProxyInfo.java

```diff
@@ -13,4 +18,12 @@ public ProxyInfo(InetSocketAddress publicAddress, InetSocketAddress sourceAddres
     this.publicAddress = publicAddress;
     this.sourceAddress = sourceAddress;
   }
+
+  public Map<String, String> toHeaders() {
+    return ImmutableMap.of(
```

Comment:
Should we allocate this in the constructor and cache in a member variable so it doesn't need to be allocated on every event?